lib/fluent/engine.rb

```diff
@@ -157,14 +158,54 @@ def run
         raise
       end
 
-      unless @log_event_verbose
-        $log.enable_event(false)
-        @fluent_log_event_router.graceful_stop
+      stop_phase(@root_agent)
+    end
+
+    # @param conf [Fluent::Config]
+    # @param supervisor [Bool]
+    # @reutrn nil
+    def reload_config(conf, supervisor: false)
+      # configure first to reduce down time while restarting
+      new_agent = RootAgent.new(log: log, system_config: @system_config)
+      ret = Fluent::StaticConfigAnalysis.call(conf, workers: system_config.workers)
+
+      ret.all_plugins.each do |plugin|
+        if plugin.respond_to?(:reloadable_plugin?) && !plugin.reloadable_plugin?
+          raise Fluent::ConfigError, "Unreloadable plugin: #{plugin.class}"
```

Comment:
This code seems good but users doesn't always concern about their using plugin implemented class.
Can we output `plugin_name` in Fluentd configuration(`@type plugin_name`)?

Reply:
fixed https://github.com/fluent/fluentd/pull/2716/commits/a78f7de7f4791faf97010be9c378f67d772b27b5